examples/terraform/basic/main.tf

```diff
@@ -0,0 +1,60 @@
+resource "talos_machine_secrets" "machine_secrets" {}
+
+resource "talos_machine_configuration_controlplane" "machineconfig_cp" {
+  cluster_name     = var.cluster_name
+  cluster_endpoint = var.cluster_endpoint
+  machine_secrets  = talos_machine_secrets.machine_secrets.machine_secrets
+}
+
+resource "talos_machine_configuration_worker" "machineconfig_worker" {
+  cluster_name     = var.cluster_name
+  cluster_endpoint = var.cluster_endpoint
+  machine_secrets  = talos_machine_secrets.machine_secrets.machine_secrets
+}
+
+resource "talos_client_configuration" "talosconfig" {
+  cluster_name    = var.cluster_name
+  machine_secrets = talos_machine_secrets.machine_secrets.machine_secrets
+  endpoints       = [for k, v in var.node_data.controlplanes : k]
+}
+
+resource "talos_machine_configuration_apply" "cp_config_apply" {
+  talos_config          = talos_client_configuration.talosconfig.talos_config
+  machine_configuration = talos_machine_configuration_controlplane.machineconfig_cp.machine_config
+  for_each              = var.node_data.controlplanes
+  endpoint              = each.key
+  node                  = each.key
+  config_patches = [
+    templatefile("${path.module}/templates/patch.yaml.tmpl", {
+      hostname     = each.value.hostname == null ? format("%s-cp-%s", var.cluster_name, index(keys(var.node_data.controlplanes), each.key)) : each.value.hostname
```

Comment:
This line was updated from https://github.com/rsmitty/terraform-provider-talos/blob/main/examples/basic/main.tf#L41, where we were looking for node_data for workers mistakenly.